assets/js/blocks/product-category/block.js

```diff
@@ -161,33 +208,39 @@ class ProductByCategoryBlock extends Component {
 				) }
 				<div className="wc-block-products-category__selection">
 					<ProductCategoryControl
-						selected={ attributes.categories }
+						selected={ currentAttributes.categories }
 						onChange={ ( value = [] ) => {
 							const ids = value.map( ( { id } ) => id );
-							setAttributes( { categories: ids } );
+							this.setAttributes( { categories: ids } );
 						} }
-						operator={ attributes.catOperator }
+						operator={ currentAttributes.catOperator }
 						onOperatorChange={ ( value = 'any' ) =>
-							setAttributes( { catOperator: value } )
+							this.setAttributes( { catOperator: value } )
 						}
 					/>
 					<Button isDefault onClick={ onDone }>
 						{ __( 'Done', 'woo-gutenberg-products-block' ) }
 					</Button>
+					<Button
+						className="wc-block-products-category__cancel-button"
+						title={ __( 'Cancel', 'woo-gutenberg-products-block' ) }
```

Comment:
Do you need the `title` prop? It's redundant for the accessibility case, since the button does contain text.